### PR TITLE
feat(compute): Stateful mig metadata example

### PR DESCRIPTION
**Pull request by askubis**
## Description

Fixes #216660851

## Checklist

**Readiness**

- [x] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [ ] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):
No, this doesn't work as per https://github.com/terraform-google-modules/terraform-docs-samples/issues/501

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):
https://cloud.google.com/compute/docs/instance-groups/configuring-stateful-metadata-in-migs

**API enablement**

- [x] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [x] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Review by msampathkumar — COMMENTED** [review]
LGTM. Lets wait for the CICD.

**Comment by msampathkumar** [comment]
Noticed following errors. These seem to unrelated. Raising a bug for the following.

```
Step #4 - "test-samples-2": TestSamples/2/sqlserver_instance_private_ip 2023-10-19T18:30:26Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; 
Step #4 - "test-samples-2": Error: Unable to remove Service Networking Connection, err: Error waiting for Delete Service Networking Connection: Error code 9, message: Failed to delete connection; Producer services (e.g. CloudSQL, Cloud Memstore, etc.) are still using this connection.


tep #3 - "test-samples-1": TestSamples/1/workflows 2023-10-19T19:07:08Z command.go:185: Error: Error creating Trigger: googleapi: Error 400: Invalid resource state for "": Permission denied while using the Eventarc Service Agent. If you recently started to use Eventarc, it may take a few minutes before all necessary permissions are propagated to the Service Agent. Otherwise, verify that it has Eventarc Service Agent role.

Step #2 - "test-samples-0": TestSamples/0/basic 2023-10-19T19:10:15Z command.go:185: Error: Error creating Trigger: googleapi: Error 400: Invalid resource state for "": Permission denied while using the Eventarc Service Agent. If you recently started to use Eventarc, it may take a few minutes before all necessary permissions are propagated to the Service Agent. Otherwise, verify that it has Eventarc Service Agent role.

Step #2 - "test-samples-0": TestSamples/0/basic 2023-10-19T19:10:15Z command.go:185:         "description": "Permission denied while using the Eventarc Service Agent. If you recently started to use Eventarc, it may take a few minutes before all necessary permissions are propagated to the Service Agent. Otherwise, verify that it has Eventarc Service Agent role."
```

File: compute/stateful_instance_group_manager/metadata/main.tf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  base_instance_name = "test"
+  target_size        = 1
+  zone               = "europe-west4-a"
+
+  version {
+    instance_template = google_compute_instance_template.default.id
+    name              = "primary"
+  }
+}
+# [START compute_stateful_instance_group_manager_metadata_pic_parent_tag]
```
Comment (msampathkumar) [inline compute/stateful_instance_group_manager/metadata/main.tf:48]:
```suggestion
# [START compute_stateful_instance_group_manager_metadata_pic]
```

File: compute/stateful_instance_group_manager/metadata/main.tf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  zone                   = google_compute_instance_group_manager.default.zone
+  name                   = "node-12"
+  preserved_state {
+    metadata = {
+      mode    = "active"
+      logging = "elaborate"
+    }
+  }
+}
+# [END compute_stateful_instance_group_manager_metadata_pic_parent_tag]
```
Comment (msampathkumar) [inline compute/stateful_instance_group_manager/metadata/main.tf:60]:
```suggestion
# [END compute_stateful_instance_group_manager_metadata_pic]
```

**Review by msampathkumar — APPROVED** [review]

**Review by gericdong — APPROVED** [review]

**Pull request merged**